frontend/src/metabase/visualizations/components/settings/ChartSettingsTableFormatting.unit.spec.js

```diff
@@ -0,0 +1,164 @@
+];
+
+const Wrapper = props => {
+  const [value, setValue] = useState([]);
+
+  return (
+    <ChartSettingsTableFormatting
+      cols={COLUMNS}
+      onChange={setValue}
+      value={value}
+      {...props}
+    />
+  );
+};
+
+const setup = props => {
+  render(<Wrapper {...props} />);
+};
+
+describe("ChartSettingsTableFormatting", () => {
+  it("Should allow you to add a rule", () => {
+    setup();
+    expect(screen.getByText("Conditional formatting")).toBeInTheDocument();
+    userEvent.click(screen.getByText("Add a rule"));
+    userEvent.click(screen.getByText("String Column"));
+    //Dismiss Popup
+    userEvent.click(screen.getByText("Which columns should be affected?"));
+
+    expect(screen.getByText("is equal to")).toBeInTheDocument();
+
+    userEvent.type(
+      screen.getByTestId("conditional-formatting-value-input"),
+      "toucan",
+    );
+    userEvent.click(screen.getByText("Add rule"));
+
+    expect(screen.getByText("String Column")).toBeInTheDocument();
+    expect(screen.getByText(/is equal to toucan/g)).toBeInTheDocument();
+  });
+
+  it("should only let you choose columns of the same type for a rule", () => {
+    setup();
+    expect(screen.getByText("Conditional formatting")).toBeInTheDocument();
+    userEvent.click(screen.getByText("Add a rule"));
+    userEvent.click(screen.getByText("String Column"));
+
+    expect(
+      screen.getByRole("option", { name: "Number Column" }),
+    ).toHaveAttribute("aria-disabled", "true");
+
+    expect(
+      screen.getByRole("option", { name: "Boolean Column" }),
+    ).toHaveAttribute("aria-disabled", "true");
+
+    expect(
+      screen.getByRole("option", { name: "String Column 2" }),
+    ).toHaveAttribute("aria-disabled", "false");
+  });
+
+  describe("should show propriate operators based on column selection", () => {
+    beforeEach(() => {
+      setup();
+      userEvent.click(screen.getByText("Add a rule"));
+    });
+
+    it("string", () => {
+      userEvent.click(screen.getByText("String Column"));
+      //Dismiss Popup
+      userEvent.click(screen.getByText("Which columns should be affected?"));
+
+      userEvent.click(
+        screen.getByTestId("conditional-formatting-value-operator-button"),
+      );
+
+      expect(
+        screen.getByRole("option", { name: "starts with" }),
+      ).toBeInTheDocument();
+
+      expect(
+        screen.getByRole("option", { name: "ends with" }),
+      ).toBeInTheDocument();
+    });
+
+    it("number", () => {
+      userEvent.click(screen.getByText("Number Column"));
+      //Dismiss Popup
+      userEvent.click(screen.getByText("Which columns should be affected?"));
+
+      userEvent.click(
+        screen.getByTestId("conditional-formatting-value-operator-button"),
+      );
+
+      expect(
+        screen.getByRole("option", { name: "is less than" }),
+      ).toBeInTheDocument();
+
+      expect(
+        screen.getByRole("option", { name: "is greater than" }),
+      ).toBeInTheDocument();
```

Comment:
Same as with strings: can we add all of the number options to make this spec more complete?

![image](https://github.com/metabase/metabase/assets/22608765/f762c069-e19b-4d4a-b2c5-182d60ba81ce)